### ISSUE TITLE
check travis build failures

### ISSUE THREAD

**Issue by jlozadad** [ANXS/postgresql#328]
seems to be failing on the older versions such as ubuntu 14.04 and centos 6 .

```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TimeoutError: Timer expired after 10 seconds
fatal: [postgresql-9.5]: FAILED! => {"changed": false, "cmd": "/bin/lsblk --list --noheadings --paths --output NAME,UUID --exclude 2", "msg": "Timer expired after 10 seconds", "rc": 257}
```

**Comment by jlozadad** [comment]
```
TASK [ANXS.postgresql : PostgreSQL | Add extensions to the databases] **********
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-9.3] (item=({u'db': u'foobar'}, u'adminpack')) => {"changed": false, "item": [{"db": "foobar"}, "adminpack"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-9.4] (item=({u'db': u'foobar'}, u'adminpack')) => {"changed": false, "item": [{"db": "foobar"}, "adminpack"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-10] (item=({u'db': u'foobar'}, u'adminpack')) => {"changed": false, "item": [{"db": "foobar"}, "adminpack"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-9.6] (item=({u'db': u'foobar'}, u'adminpack')) => {"changed": false, "item": [{"db": "foobar"}, "adminpack"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-9.5] (item=({u'db': u'foobar'}, u'adminpack')) => {"changed": false, "item": [{"db": "foobar"}, "adminpack"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-9.3] (item=({u'db': u'foobar'}, u'pgcrypto')) => {"changed": false, "item": [{"db": "foobar"}, "pgcrypto"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-9.4] (item=({u'db': u'foobar'}, u'pgcrypto')) => {"changed": false, "item": [{"db": "foobar"}, "pgcrypto"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-10] (item=({u'db': u'foobar'}, u'pgcrypto')) => {"changed": false, "item": [{"db": "foobar"}, "pgcrypto"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-9.5] (item=({u'db': u'foobar'}, u'pgcrypto')) => {"changed": false, "item": [{"db": "foobar"}, "pgcrypto"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-9.6] (item=({u'db': u'foobar'}, u'pgcrypto')) => {"changed": false, "item": [{"db": "foobar"}, "pgcrypto"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-9.3] (item=({u'db': u'foobar'}, u'unaccent')) => {"changed": false, "item": [{"db": "foobar"}, "unaccent"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-9.4] (item=({u'db': u'foobar'}, u'unaccent')) => {"changed": false, "item": [{"db": "foobar"}, "unaccent"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-10] (item=({u'db': u'foobar'}, u'unaccent')) => {"changed": false, "item": [{"db": "foobar"}, "unaccent"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-9.6] (item=({u'db': u'foobar'}, u'unaccent')) => {"changed": false, "item": [{"db": "foobar"}, "unaccent"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [postgresql-9.5] (item=({u'db': u'foobar'}, u'unaccent')) => {"changed": false, "item": [{"db": "foobar"}, "unaccent"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
	to retry, use: --limit @/home/travis/build/ANXS/postgresql/tests/docker/site.retry
```

**Comment by gclough** [comment]
@WolfgangPecho ... can you help out here?  How did you fix this in #267?

**Comment by WolfgangPecho** [comment]
@gclough I found [https://groups.google.com/forum/#!msg/ansible-project/lS0ifitbltY/55KA4RrXAgAJ](url). Looks like there can be problems uslng lsblk. 
It strikes me that ubuntu-14.04 and centos-6 are SysV-Init based. May be it is a timing issue mounting the file systems.  I (with very limited knowledge in that field :-)) think at the level of ansible we have no chance to fix it

**Comment by gclough** [comment]
I created a clean local Ubuntu 14.04 VM, and on the first install it fails with "*peer authentication*":

```
TASK [postgresql : PostgreSQL | Add extensions to the databases] ***************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [192.168.64.175] (item=({u'db': u'foobar'}, u'adminpack')) => {"changed": false, "item": [{"db": "foobar"}, "adminpack"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [192.168.64.175] (item=({u'db': u'foobar'}, u'pgcrypto')) => {"changed": false, "item": [{"db": "foobar"}, "pgcrypto"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  Peer authentication failed for user "postgres"
failed: [192.168.64.175] (item=({u'db': u'foobar'}, u'unaccent')) => {"changed": false, "item": [{"db": "foobar"}, "unaccent"], "msg": "unable to connect to database: FATAL:  Peer authentication failed for user \"postgres\"\n"}
	to retry, use: --limit @/home/ansible/postgres.retry
```

It seems that the cluster isn't restarted/reloaded after the config changes... so the `pg_hba.conf` isn't read after "trust" is added.  I restarted it manually, and on the second try it fails wit "*missing database*"h:

```
TASK [postgresql : PostgreSQL | Add extensions to the databases] ***************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  database "foobar" does not exist
failed: [192.168.64.175] (item=({u'db': u'foobar'}, u'adminpack')) => {"changed": false, "item": [{"db": "foobar"}, "adminpack"], "msg": "unable to connect to database: FATAL:  database \"foobar\" does not exist\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  database "foobar" does not exist
failed: [192.168.64.175] (item=({u'db': u'foobar'}, u'pgcrypto')) => {"changed": false, "item": [{"db": "foobar"}, "pgcrypto"], "msg": "unable to connect to database: FATAL:  database \"foobar\" does not exist\n"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OperationalError: FATAL:  database "foobar" does not exist
failed: [192.168.64.175] (item=({u'db': u'foobar'}, u'unaccent')) => {"changed": false, "item": [{"db": "foobar"}, "unaccent"], "msg": "unable to connect to database: FATAL:  database \"foobar\" does not exist\n"}
	to retry, use: --limit @/home/ansible/postgres.retry
```

**Comment by gclough** [comment]
@WolfgangPecho ... this is weird.  If Ubuntu 14.04 and CentOS 6 are similarly based... then why does CentOS work. :-/

**Comment by gclough** [comment]
My gut feel is that it's got something to do with the restart being moved to a handler...

**Comment by gclough** [comment]
Here's the problem:

```
- name: PostgreSQL | Reset the cluster - create a new one (with specified encoding and locale) | Debian
  shell: >
    pg_createcluster --start --locale {{ postgresql_locale }}
```

That starts the cluster immediately... and critically, *BEFORE* the `pg_hba.conf` has been updated with "trust"

```
       --start
           Immediately start a server for the cluster after creating it (i. e.
           call pg_ctlcluster version cluster start on it). By default, the
           cluster is not started.
```

**Comment by WolfgangPecho** [comment]
@gclough Could you tell me how you created your vm with ubuntu-14.04 ?
I cannot reproduce the error on my local setup ( vagrant with bento/ubuntu-14.04 or the official ubuntu ubuntu/trusty64 ) using the master branch.

**Comment by gclough** [comment]
I just downloaded Ubuntu ISO, and built a new VM.   I've tried to get Vagrant and Docker working... but with no success.  The good thing is that I think I've cracked it, as we just need to do a reload to enable the `pg_hba.conf` changes.

It's probable that in earlier versions of Ubuntu, PostgreSQL is started by default... so it doesn't pick up any of the parameter changes we do.  Adding a reload fixes most of those, but won't pick up any of the major parameter changes that require a re*start*.

I've created #291 to fix that.
